src/main/resources/dbreplica/migration/V1.0.19__refresh_report_tasks.sql

```diff
@@ -0,0 +1,41 @@
+CREATE OR REPLACE PROCEDURE cft_task_db.refresh_report_tasks(
+  max_rows_to_process INTEGER)
+  LANGUAGE plpgsql
+AS $procedure$
+
+declare
+l_task_id_list              TEXT[];
+l_task_id                   TEXT;
+l_count_of_records          INTEGER;
+l_records_to_process        INTEGER;
+
+begin
+
+    select count(task_id) FROM cft_task_db.tasks where report_refresh_request_time is not null
+    into STRICT l_count_of_records;
+
+    if (l_count_of_records > 0) then
```

Comment:
Very well thought out. Guards against when max_rows_to_process < count_of_records. Also covered in parameterized test.